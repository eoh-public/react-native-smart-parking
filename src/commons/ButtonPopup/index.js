import React from 'react';
import { StyleSheet, View } from 'react-native';
import Modal from 'react-native-modal';
import { Icon } from '@ant-design/react-native';

import { Colors, Device } from '../../configs';
import { TESTID } from '../../configs/Constants';
import Text from '../../commons/Text';

import BottomButtonView from '../BottomButtonView';

const ButtonPopup = ({
  rowButton = true,
  visible,
  children,
  mainTitle,
  secondaryTitle,
  thirdTitle,
  onClose,
  onPressMain,
  onPressSecondary,
  onPressThird,
  bodyStyle,
  hideClose,
  typeMain,
  typeSecondary,
  typeThird,
  semiboldSecond,
  title,
  childrenStyle,
  titleStyle,
  bottomStyles,
}) => {
  return (
    <Modal
      isVisible={visible}
      onBackButtonPress={onClose}
      onBackdropPress={onClose}
      testID={TESTID.MODAL_BUTTON_POPUP}
    >
      <View style={styles.popoverStyle}>
        <View style={styles.modalWrapper}>
          <View style={titleStyle}>
            {title && (
              <Text type="H4" bold>
                {title}
              </Text>
            )}
            {!hideClose && (
              <Icon
                name={'close'}
                color={Colors.Gray8}
                style={styles.close}
                onPress={onClose}
              />
            )}
          </View>
          <View
            style={[
              childrenStyle ? childrenStyle : styles.modalHeader,
              bodyStyle,
            ]}
          >
            {children}
          </View>
          <BottomButtonView
            rowButton={rowButton}
            style={[
              styles.bottomButtonView,
              thirdTitle && styles.thirdStyle,
              bottomStyles,
            ]}
            secondaryTitle={secondaryTitle}
            mainTitle={mainTitle}
            onPressSecondary={onPressSecondary}
            onPressMain={onPressMain}
            typeMain={typeMain}
            typeSecondary={typeSecondary}
            testIDPrefix={TESTID.PREFIX.BUTTON_POPUP}
            semiboldSecond={semiboldSecond}
          />
          {thirdTitle && (
            <BottomButtonView
              style={styles.bottomButtonView2}
              mainTitle={thirdTitle}
              onPressMain={onPressThird}
              typeMain={typeThird}
            />
          )}
        </View>
      </View>
    </Modal>
  );
};

export default ButtonPopup;

const styles = StyleSheet.create({
  popoverStyle: {
    width: '100%',
    backgroundColor: Colors.White,
    position: 'absolute',
    borderRadius: 10,
  },
  modalWrapper: {
    flex: 1,
    flexDirection: 'column',
    position: 'relative',
    backgroundColor: Colors.White,
    borderRadius: 10,
    marginBottom: Device.isIphoneX ? 20 : 0,
  },
  modalHeader: {
    padding: 16,
    paddingTop: 60,
  },
  close: {
    position: 'absolute',
    top: 20,
    right: 20,
    zIndex: 1,
  },
  bottomButtonView: {
    position: 'relative',
    marginBottom: 24,
  },
  bottomButtonView2: {
    position: 'relative',
    marginBottom: 24,
  },
  thirdStyle: {
    marginBottom: 0,
  },
});
