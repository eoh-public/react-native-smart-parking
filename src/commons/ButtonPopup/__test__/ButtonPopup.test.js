import React from 'react';
import renderer, { act } from 'react-test-renderer';
import ButtonPopup from '../index';
import Modal from 'react-native-modal';

describe('Test button popup', () => {
  let tree;

  test('create button popup', () => {
    act(() => {
      tree = renderer.create(
        <ButtonPopup
          mainTitle="Main title"
          secondaryTitle="Second title"
          thirdTitle="thirdTitle"
        />
      );
    });
    const instance = tree.root;
    const modal = instance.findAllByType(Modal);
    expect(modal.length).toBe(1);
  });
});
