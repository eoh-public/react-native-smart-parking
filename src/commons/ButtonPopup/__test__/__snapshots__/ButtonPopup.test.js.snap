// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button popup create button popup 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="MODAL_BUTTON_POPUP"
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "flex": 1,
            "flexDirection": "column",
            "marginBottom": 20,
            "position": "relative",
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#595959",
                  "fontFamily": "antoutline",
                  "fontSize": 22,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "position": "absolute",
                  "right": 20,
                  "top": 20,
                  "zIndex": 1,
                },
              ]
            }
          >
            î©Œ
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "padding": 16,
                "paddingTop": 60,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "flex": 1,
                "left": 0,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "position": "absolute",
                "right": 0,
              },
              Object {
                "flexDirection": "row-reverse",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "marginBottom": 24,
                  "position": "relative",
                },
                Object {
                  "marginBottom": 0,
                },
              ],
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00979D",
                "borderRadius": 28,
                "borderWidth": 0,
                "flex": 1,
                "height": 48,
                "justifyContent": "center",
                "marginLeft": 4,
                "opacity": 1,
                "width": "100%",
              }
            }
            testID="BUTTON_POPUPBOTTOM_VIEW_MAIN"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {},
                  ]
                }
              >
                Main title
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E8E8E8",
                "borderRadius": 28,
                "borderWidth": 1,
                "flex": 1,
                "height": 48,
                "justifyContent": "center",
                "marginRight": 4,
                "opacity": 1,
                "width": "100%",
              }
            }
            testID="BUTTON_POPUPBOTTOM_VIEW_SECONDARY"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#595959",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {},
                  ]
                }
              >
                Second title
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "flex": 1,
                "left": 0,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "position": "absolute",
                "right": 0,
              },
              false,
              Object {
                "marginBottom": 24,
                "position": "relative",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00979D",
                "borderRadius": 28,
                "borderWidth": 0,
                "flex": 1,
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
            testID="BOTTOM_VIEW_MAIN"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {},
                  ]
                }
              >
                thirdTitle
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
